HopsanGUI/GUIObjects/GUIComponent.cpp

```diff
@@ -139,8 +139,60 @@ QString Component::getTypeCQS() const
 //! @brief Set a parameter value to be mapped to a System parameter
 bool Component::setParameterValue(QString name, QString value, bool force)
 {
+    if(this->getTypeName() == "FMIWrapper" && name == "path") {
+        //Remove old ports
+        QList<Port*> ports = this->getPortListPtrs();
+        for(const auto port : ports) {
+            this->removeExternalPort(port->getName());
+        }
+        this->getAppearanceData()->getPortAppearanceMap().clear();
+    }
+
     bool retval =  mpParentSystemObject->getCoreSystemAccessPtr()->setParameterValue(this->getName(), name, value, force);
 
+    if(this->getTypeName() == "FMIWrapper" && name == "path") {
+        //Get lists of input and output ports from core component
+        QStringList inputs, outputs;
+        for(const auto &port: mpParentSystemObject->getCoreSystemAccessPtr()->getPortNames(this->getName())) {
+            QString type = mpParentSystemObject->getCoreSystemAccessPtr()->getPortType(this->getName(), port);
+            SharedPortAppearanceT app(new PortAppearance());
+            if("ReadPortType" == type) {
+                inputs << port;
+            }
+            else if("WritePortType" == type) {
+                outputs << port;
+            }
+        }
+
+        //Auto-place input ports
+        for(int i=0; i<inputs.size(); ++i) {
+            SharedPortAppearanceT app(new PortAppearance());
```

Comment:
You are right, there is a `refreshExternalPortsAppearanceAndPosition()` function. It is only available in `SystemObject` though. Perhaps we can move this function to `ModelObject` and use it also for FMI components. I will have a look.

Reply:
yes or we can add it later

Reply:
Yes, that code was quite different as well. It looks on the position of the internal model objects in the container and places the ports based on this. But perhaps some parts of it can still be abstracted out and re-used.